dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/converter/EnrollmentTrackerConverterService.java

```diff
@@ -127,10 +127,13 @@ public ProgramInstance from( TrackerPreheat preheat, Enrollment enrollment )
         List<ProgramInstance> programInstances = new ArrayList<>();
 
         enrollments.forEach( enrollment -> {
+
             ProgramInstance programInstance = preheat.getEnrollment( TrackerIdScheme.UID, enrollment.getEnrollment() );
-            OrganisationUnit organisationUnit = preheat.get( TrackerIdScheme.UID, OrganisationUnit.class, enrollment.getOrgUnit() );
+            OrganisationUnit organisationUnit = preheat
+                .get( TrackerIdScheme.UID, OrganisationUnit.class, enrollment.getOrgUnit() );
             Program program = preheat.get( TrackerIdScheme.UID, Program.class, enrollment.getProgram() );
-            TrackedEntityInstance trackedEntityInstance = preheat.getTrackedEntity( TrackerIdScheme.UID, enrollment.getTrackedEntity() );
+            TrackedEntityInstance trackedEntityInstance = preheat
```

Comment:
This method call may return `null`. Is that ok that we set a `null` value in `programInstance`?

Reply:
I think this is a very good question.
The validation is supposed to make sure it don't happen.
I would argue that this is not ok and we should call an assert/np error here to make it very clear we don't allow persistence with out this critical references.

Reply:
I have added Objects.nonNull checks. Let's discuss this in next meeting.